In [1]:
defaultTime:1000

(defaultTime:1000)

In [2]:
import : dir co : up : up : .

In [4]:
count : property nilpotent {rev:B} : S5?
count : property nilpotent {null:B} : S5?
count : property nilpotent_right {null:B} : S5?
count : property nilpotent_right {A A} : S5?

1
0
0
1

In [ ]:
Property nilpotent (:{rev:B}) (:{null:B}) : S5? 

In [7]:
aq {count:property nilpotent A:B} {rev:B} {null:B} : S5? 

1 0

In [9]:
{aq 2 property nilpotent {rev:B} {null:B}:B} : S5?

1 0

Search for nilpotents from C0, basic commands.
First level: use the S6 sample.

In [ ]:
Def Property : ap {(bin (get:B) : property nilpotent (get:B) : S

In [ ]:
property nilpotent {rev:B} : S6?

In [ ]:
ap {(bin (get:B) : property nilpotent (get:B) : S6?)} : (:{rev:B}) 

In [ ]:
Def nilS6 : ap {(bin (get:B) : property nilpotent (get:B) : S6?)}

In [ ]:
nilS6 : (:{rev:B}) (:{null:B}) 

In [ ]:
defaultTime:1000

In [ ]:
out C0-S6 3600 : nilS6 : C0?:

In [ ]:
stat : C0-S6

In [ ]:
count : in stable : C0-S6

In [ ]:
out C2-S6.sample 3600 : nilS6 : first 100 : C2? :

In [ ]:
stat : C2-S6.sample 

In [ ]:
in stable : C2-S6.sample

In [15]:
Let Cand : in : C0-S6.sample.candidates

In [16]:
Cand?

(:has {domain : B}) (:null {B get}) (:{nif : B} b) (:put {const : B}) (:null {B pass}) (:if {prod : B}) (:has {B}) (:is {b : B}) (:get {B : A}) (:{null : B} a) (:null sum) (:if {B : plus}) (:null {B : hasnt}) (:{A : B} is) (:null {domain B}) (:null {right B}) (:get pass) (:get {A : B}) (:null {prod : B}) (:null {null : B}) (:null {1 : B}) (:is {const : B}) (:is is) (:is {B has}) (:has {B : b}) (:{sum : B} null) (:null {B 0}) (:has {hasnt B}) (:is {nif : B}) (:null {B prod}) (:null {has B}) (:null {get B}) (:if {a B}) (:has {prod B}) (:get {B : bin}) (:is {B : sum}) (:has {B b}) (:null {B : arg}) (:null {0 : B}) (:null b) (:if put) (:if {is : B}) (:is {has B}) (:{is : B} bin) (:get {B isnt}) (:null if) (:{sum : B} star) (:null {B arg}) (:is {B sum}) (:if {B isnt}) (:{get : B} plus) (:get {B : left}) (:has has) (:if {right B}) (:null {b B}) (:null hasnt) (:is {B const}) (:null {b : B}) (:const) (:is {domain B}) (:has {B domain}) (:has {B : arg}) (:{is : B} put) (:if null) (:null {B : is}

In [12]:
Def nilS53 : ap {(bin (get:B) : property nilpotent (get:B) : S5? S3?)}

In [18]:
out C2-S6-S53 30000 : nilS53 : Cand?

eval> creation, max seconds:30000.0, max GB:2.0
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:100, max GB:2
eval> creation, max seconds:

In [4]:
eval 1000 : count : ap {nif ((right:B)=0):B} : in : C2-S6-S53

8

In [3]:
eval 1000 : Let C2-S6-S53-cand : ap {if ((right:B)=0):B} : in : C2-S6-S53

In [5]:
count : C2-S6-S53-cand?

141

In [9]:
import : dir co : up : up : .

In [8]:
defaultTime:1000

(defaultTime:1000)

In [12]:
count : C2-S6-S53-cand?

141

In [13]:
Def tab : ap post <\n>

In [18]:
tab : first 10 : C2-S6-S53-cand?

(bin has {domain : B}:0) 
 (bin null {B get}:0) 
 (bin {nif : B} b:0) 
 (bin put {const : B}:0) 
 (bin null {B pass}:0) 
 (bin if {prod : B}:0) 
 (bin has {B}:0) 
 (bin is {b : B}:0) 
 (bin get {B : A}:0) 
 (bin {null : B} a:0) 


In [20]:
Let Candidates : ap {put : arg : B} : C2-S6-S53-cand?

In [21]:
count : Candidates?

141

The 141 candidates are nilpotent with respect to S6? S5? S3? 

Let's find those which are also nilpotent_right.  If data is both nilpotent_left and nilpotent_right, it is a candidate ring.  I already know that these are multiplies of some fixed constant, but I'm going through the motions and an exercise. 

In [25]:
property nilpotent {first:B} : S5?

21

In [26]:
Def nil_right_S53 : ap {(bin (get:B) : property nilpotent_right (get:B) : S5? S3?)}

In [32]:
property nilpotent_right get {B : A} : S6? S3?

0

In [33]:
S6?

(:a) (:b) (:a b) ◎ (:1) (:2) (:bin) (:(bin a:x)) (:(bin b:y))

In [37]:
property nilpotent_right {A A} : S6?

8